### PR TITLE
Pointwise scheduler rework

### PR DESCRIPTION
**Pull request by csarofeen**

### CODE REVIEW

File: torch/csrc/jit/codegen/cuda/scheduler/registry.cpp [hunk 1]
```diff
@@ -9,12 +9,15 @@ namespace fuser {
...
-  if (has_param_ != other->has_param_) {
+  if (has_reduction_param_ != other->has_reduction_param_) {
     return false;
   }
-  if (has_param_) {
+  if (has_reduction_param_) {
     return rparams_ == other->rparams_;
+  } else {
+    return pparams_ == other->pparams_;
   }
```
Comment (shmsong) [inline torch/csrc/jit/codegen/cuda/scheduler/registry.cpp:19]: This makes sure the caching is safe. We may want to update the hash as well. It's `size_t SchedulerEntryHash::operator()(const SchedulerEntry& se) const {` at around line 362.

**Review by shmsong — COMMENTED** [review]
Overall looks good to me. Left a minor comment on appending `PointWiseParams` to the hash.

File: torch/csrc/jit/codegen/cuda/scheduler/registry.cpp (hunk 1 above)
Comment (csarofeen) [reply torch/csrc/jit/codegen/cuda/scheduler/registry.cpp:19]: Just fixed this, will push. Thanks.

**Comment by csarofeen** [comment]
Closed in favor of https://github.com/csarofeen/pytorch/pull/814

**Pull request closed without merging**